vectorstores/cloudsql/vectorstore.go

```diff
@@ -39,3 +41,72 @@ type SearchDocument struct {
 }
 
 var _ vectorstores.VectorStore = &VectorStore{}
+
+// NewVectorStore creates a new VectorStore with options.
+func NewVectorStore(ctx context.Context, engine cloudsqlutil.PostgresEngine, embedder embeddings.Embedder, tableName string, opts ...CloudSQLVectoreStoresOption) (VectorStore, error) {
+	vs, err := applyCloudSQLVectorStoreOptions(engine, embedder, tableName, opts...)
+	if err != nil {
+		return VectorStore{}, err
+	}
+	return vs, nil
+}
+
+// ApplyVectorIndex creates an index in the table of the embeddings
+func (vs *VectorStore) ApplyVectorIndex(ctx context.Context, index BaseIndex, name string, concurrently, overwrite bool) error {
+	if index.indexType == "exactnearestneighbor" {
+		return vs.DropVectorIndex(ctx, name, overwrite)
+	}
+	function := index.distanceStrategy.searchFunction()
+	filter := ""
+	if len(index.partialIndexes) > 0 {
+		filter = fmt.Sprintf("WHERE %s", index.partialIndexes)
+	}
+	optsString, err := index.indexOptions()
+	if err != nil {
+		return fmt.Errorf("indexOptions error: %w", err)
+	}
+	params := fmt.Sprintf("WITH %s", optsString)
+
+	if name == "" {
+		if index.name == "" {
+			index.name = vs.tableName + defaultIndexNameSuffix
+		}
+		name = index.name
+	}
+
+	concurrentlyStr := ""
+	if concurrently {
+		concurrentlyStr = "CONCURRENTLY"
+	}
+
+	stmt := fmt.Sprintf("CREATE INDEX %s %s ON %s.%s USING %s (%s %s) %s %s",
+		concurrentlyStr, name, vs.schemaName, vs.tableName, index.indexType, vs.embeddingColumn, function, params, filter)
+
+	_, err = vs.engine.Pool.Exec(ctx, stmt)
+	if err != nil {
+		return fmt.Errorf("failed to execute creation of index: %w", err)
+	}
+	_, err = vs.engine.Pool.Exec(ctx, "COMMIT")
+	if err != nil {
+		return fmt.Errorf("failed to commit index: %w", err)
+	}
+	return nil
+}
+
+// DropVectorIndex drops the vector index from the VectorStore.
+func (vs *VectorStore) DropVectorIndex(ctx context.Context, indexName string, overwrite bool) error {
+	// Overwrite allows dangerous operations like a Drop query.
+	if !overwrite {
+		return nil
+	}
```

Comment:
Hmm this seems odd. If they don't overwrite we do nothing?

Reply:
you are right! this is not on the python version, removing it!